### DISCUSSION TITLE
Why Analysis kwarg is called datas instead of data?

### DISCUSSION THREAD

**Discussion by GMaxera** [pyinstaller/pyinstaller#7193] [PyInstaller]
From the documentation and various examples:

> datas: non-binary files included in the app, including names given by the [--add-data](https://pyinstaller.org/en/stable/usage.html#cmdoption-add-data) option.

```
a = Analysis(
    ["script.py"],
    pathex=[],
    binaries=[],
    datas=data,
```

but `datas` does not exist has English word, it's just wrong. Data is the plural of datum and it's used as mass noun also.
So, there is no datas in the English dictionary: https://www.wordreference.com/definition/data

So, I was wondering what was the rationale behind calling it `datas`.

Thanks.

**Comment by rokm** [comment]
Presumably to make it clear that it is about data entries (analogue to binaries' entries and hidden import entries) and not about generic data, because "data" in that context could mean pretty much anything. If you want a definite answer, you'll have to ask people who were in charge 15 years ago when the argument was introduced. And if your question is why it remained that way until now, the answer is backward compatibility with existing spec files.

**Reply by bwoodsend** [reply]
A list of directories to prepend to `sys.path` is called _pathex_. PyIntaller, which doesn't actually install anything - it just collects stuff, is still called PyInstaller. I wouldn't read too much into the names of things here.

**Reply by GMaxera** [reply]
🤔 I have to be frank, I'm quite disappointed by your answers. It seems you are saying that there is no reasons at all, and it seems you don't put much attention to typos or made up words 🤔 .
It's still possible to keep backward compatibility on the interface, and still move forward and phase out typos and non-sense words.
I don't see why not.

**Reply by bwoodsend** [reply]
Is it really worth breaking millions of projects just to appease some funky grammar?

**Reply by GMaxera** [reply]
You don't have to break millions of projects. Just adding both data and datas as kwargs and documenting to use "data" instead of datas will gradually phase out "datas".

Also, if I would be an English native speaker, I would be offended from you labelling "funky" my own language grammar.

Anyway, you can close this discussion.
I initially thought that there was some reasons behind I couldn't see, but now seems obvious that it's just a grammatical mistake that remains for "historical reasons" 🤷‍♂️ 

Thanks and sorry for have took your time on dealing with this.
